specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-06-01-preview/integrationRuntime.json

```diff
@@ -2470,6 +2532,11 @@
           "type": "integer",
           "format": "int32",
           "readOnly": true
+        },
+        "selfContainedInteractiveAuthoringEnabled": {
```

Comment:
>selfContainedInteractiveAuthoringEnabled [](http://example.com/codeflow?start=9&length=40)

How about this is enum as selfContainedInteractiveAuthoringState with possible values as Enabled, Disabled, NotApplicable etc.

------------- Guidance ---------------

ARM recommends enums over booleans for future proof APIs. 

 

Standard guidance is: replace boolean/switch properties with a more meaningful enum whenever possible. 

A boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated]. 

Note: do NOT define a 'boolean enum' with two values 'True and False'. This might be easier to 'extend' in terms of types, but semantically its cofusing, and no better than a boolean.